### ISSUE TITLE
Figure out if all other GET endpoints are still useful in the legacy API

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#1721]
Tracker: #1680 

The legacy API exposes a GET API to fetch versions attached to an add-on and a few other things in the "2" api version. We should list those endpoints and figure out if they are still useful.

**Comment by diox** [comment]
TL;DR: anything behind the "2" API version appears to be unused and undocumented.

The following endpoints are behind API version "2":
- https://services.addons.mozilla.org/api/2/user/*
- https://services.addons.mozilla.org/api/2/addons/
- https://services.addons.mozilla.org/api/2/addon/[id]/versions/
- https://services.addons.mozilla.org/api/2/addon/[id]/version/[version_id]/

There is some overlap with #1715 as most are useless without auth, returning an oauth error or a 404.

AFAIK none of those are used by Firefox (I could not find references to these APIs in about:config, and [AddonRepository.jsm](https://dxr.mozilla.org/mozilla-central/source/toolkit/mozapps/extensions/internal/AddonRepository.jsm) sets `API_VERSION` to `1.5` anyway).

They are also undocumented: [MDN documentation for Add-ons manager API](https://developer.mozilla.org/en-US/Add-ons/AMO/Add-ons_manager_API) says that the latest API version is 1.5 and does not document them anyway.
